### DISCUSSION TITLE
Any way to make a uniform vec3 display in the UI as a group of numbers rather than assuming it's a color?

### DISCUSSION THREAD

**Discussion by Shnazzy** [bnpr/Malt#500] [Q&A] [answered]
I'm opting to use GLSL code instead of nodes.  My shader has a vec3 uniform, but it's not a color, and is instead just a vector in 3D space.  Malt seems convinced though that it's a color, and displays it like this on the UI panel.  However, if I make an equivalent of this shader but with nodes and declare the vec3 as a property, it shows up the way I'd want it with three individual numbers.  Is there some kind of way to mark uniforms in my code to use a preferred subtype?  I had originally assumed the META customization options would make this happen but if they can, I clearly haven't figured out the right way.

While looking through the code I think I saw a TODO related to this (if my assumptions are right), but I'm not sure how much of a priority it is.

Thanks.

**Comment by pragma37** [comment, marked as answer]
This is not possible at the moment and is not currently planned (I'm not sure what TODO you are referring to).

You can always use a single node in your node tree if that's what you want.
Also, keep in mind that some features, like AOVs, are only available through nodes.
